allennlp/data/fields/multilabel_field.py

```diff
@@ -0,0 +1,130 @@
+        integers for you (e.g., "entailment" and "contradiction" get converted to 0, 1, ...),
+        and this namespace tells the ``Vocabulary`` object which mapping from strings to integers
+        to use (so "entailment" as a label doesn't get the same integer id as "entailment" as a
+        word).  If you have multiple different label fields in your data, you should make sure you
+        use different namespaces for each one, always using the suffix "labels" (e.g.,
+        "passage_labels" and "question_labels").
+    skip_indexing : ``bool``, optional (default=False)
+        If your labels are 0-indexed integers, you can pass in this flag, and we'll skip the indexing
+        step.  If this is ``False`` and your labels are not strings, this throws a ``ConfigurationError``.
+    num_labels : ``int``, optional (default=None)
+        If ``skip_indexing=True``, the total number of possible labels should be provided, which is needed
+        in deciding the size of the output tensor. `num_labels` should equal largest label id + 1
+        If ``skip_indexing=False``, it should remain None
+
+    """
+    # It is possible that users want to use this field with a namespace which uses OOV/PAD tokens.
+    # This warning will be repeated for every instantiation of this class (i.e for every data
+    # instance), spewing a lot of warnings so this class variable is used to only log a single
+    # warning per namespace.
+    _already_warned_namespaces: Set[str] = set()
+
+    def __init__(self,
+                 labels: Sequence[Union[str, int]],
+                 label_namespace: str = 'labels',
+                 skip_indexing: bool = False,
+                 num_labels: Optional[int] = None) -> None:
+        self.labels = labels
+        self._label_namespace = label_namespace
+        self._label_ids = None
+        self._maybe_warn_for_namespace(label_namespace)
+        self._num_labels = num_labels
+
+        if skip_indexing:
+            if not all(isinstance(label, int) for label in labels):
+                raise ConfigurationError("In order to skip indexing, your labels must be integers. "
+                                         "Found labels = {}".format(labels))
+            if not num_labels:
+                raise ConfigurationError("In order to skip indexing, num_labels can't be None.")
+
+            if not all(cast(int, label) < num_labels for label in labels):
+                raise ConfigurationError("All labels should be < num_labels. "
+                                         "Found num_labels = {} and labels = {} ".format(num_labels, labels))
+
+            self._label_ids = labels
+        else:
+            if not all(isinstance(label, str) for label in labels):
+                raise ConfigurationError("MultiLabelFields expects string labels if skip_indexing=False. "
+                                         "Found labels: {}".format(labels))
+            if num_labels:
```

Comment:
This isn't necessary - just add an `if self._num_labels is not None:` block in `index`, because then you could also pass the number of labels if you know it.